message_ix_models/model/water/report.py

```diff
@@ -271,6 +278,152 @@ def multiply_electricity_output_of_hydro(
     return report_iam
 
 
+def pop_water_access(sc: Scenario, reg: str, sdgs: bool = False) -> None:
```

Comment:
```suggestion
def pop_water_access(sc: Scenario, reg: str, sdgs: bool = False) -> pd.DataFrame:
```

Not sure how mypy accepted this, but the function clearly returns a `pd.DataFrame` and not `None`. Is mypy enabled for the water module?

Reply:
It is not: https://github.com/iiasa/message-ix-models/blob/a45fce49fe3aa17ddb9d83f8e5b550accc0090b4/pyproject.toml#L113-L117

Reply:
Thanks for checking, that's a shame :neutral_face:

Reply:
With a new colleague (Vignesh) coming onboard to work on nexus/water topics, perhaps some of the added bandwidth could be reserved for improving code quality through that and other TODOs.

Reply:
thanks, but in theory, if I have the extension on VSC should it do my typewriting itself?
I have mypy type checker (the open source mypy breaks..)

Reply:
I'm also using the mypy type checker extension :)
But I don't see anywhere in their docs that they're automatically adding types to something. I do also have the Pylance extension and this infers some types even for untyped functions in the signature help it displays when hovering over a function name (for example). However, this does not mean that these functions are type checked! Pylance is just guessing the types and nothing makes sure these guesses are enforced. So if you want your functions to have type hints (for all the many good reasons), you have to add them manually, I think. 
But as I mentioned before: you already spell out which types the parameters should have in the docstring, so it's not a lot of work :)